### PR TITLE
0063 Move examples for internal functions to unit tests

### LINKED ISSUE #63: Examples for unexported functions
As the first submission to CRAN approaches, and given that we have example code in the Examples section of unexported functions, as per CRAN requirements, we will have either to omit these examples or to export these functions.

I think we should try to convert those examples to unit tests, and remove the examples, while leaving the unexported functions as such.

Here is the output of a quick grep on `"sdtm.oak:::"`:

```
cnd_df.R:#' sdtm.oak:::new_cnd_df(dat = df, cnd = c(FALSE, NA, TRUE))
cnd_df.R:#' sdtm.oak:::is_cnd_df(df)
cnd_df.R:#' cnd_df <- sdtm.oak:::new_cnd_df(dat = df, cnd = c(FALSE, NA, TRUE))
cnd_df.R:#' sdtm.oak:::is_cnd_df(cnd_df)
cnd_df.R:#' sdtm.oak:::get_cnd_df_cnd(df)
cnd_df.R:#' cnd_df <- sdtm.oak:::new_cnd_df(dat = df, cnd = c(FALSE, NA, TRUE))
cnd_df.R:#' sdtm.oak:::get_cnd_df_cnd(cnd_df)
cnd_df.R:#' sdtm.oak:::get_cnd_df_cnd_sum(df)
cnd_df.R:#' cnd_df <- sdtm.oak:::new_cnd_df(dat = df, cnd = c(FALSE, NA, TRUE))
cnd_df.R:#' sdtm.oak:::get_cnd_df_cnd_sum(cnd_df)
cnd_df.R:#' cnd_df <- sdtm.oak:::new_cnd_df(dat = df, cnd = c(FALSE, NA, TRUE))
cnd_df.R:#' sdtm.oak:::is_cnd_df(cnd_df)
cnd_df.R:#' sdtm.oak:::is_cnd_df(sdtm.oak:::rm_cnd_df(cnd_df))
cnd_df.R:#' sdtm.oak:::eval_conditions(df, x > 2)
cnd_df.R:#' sdtm.oak:::eval_conditions(df, x > 2 & y)
cnd_df.R:#' sdtm.oak:::eval_conditions(df, x > 2, y)
cnd_df.R:#' sdtm.oak:::eval_conditions(df_with_na, x > 2, .na = FALSE)
cnd_df.R:#' sdtm.oak:::eval_conditions(df, x > w)
cnd_df.R:#' sdtm.oak:::eval_conditions(df, x > w, .env = env)
cnd_df.R:#' sdtm.oak:::eval_conditions(df, x > w, .env = list(w = 3))
cnd_df.R:#' sdtm.oak:::eval_conditions(df, x > w, .env = tibble::tibble(w = 4))
ct.R:#' all.equal(ct_spec_01, sdtm.oak:::assert_ct_spec(ct_spec_01))
ct.R:#' # assertion; `sdtm.oak:::ct_spec_vars()` defines their names.
ct.R:#' (req_vars <- sdtm.oak:::ct_spec_vars())
ct.R:#' sdtm.oak:::assert_ct_spec(ct_spec_01[req_vars])
ct.R:#' try(sdtm.oak:::assert_ct_spec(ct_spec_01[opt_vars]))
ct.R:#' sdtm.oak:::assert_ct_clst(ct_spec = ct_spec, ct_clst = "C71113")
ct.R:#' sdtm.oak:::assert_ct_clst(ct_spec = ct_spec, ct_clst = NULL, optional = TRUE)
ct.R:#' try(sdtm.oak:::assert_ct_clst(ct_spec = ct_spec, ct_clst = NULL, optional = FALSE))
ct.R:#' sdtm.oak:::ct_mappings(ct_spec = ct_spec_01)
ct.R:#' sdtm.oak:::ct_mappings(ct_spec = ct_spec_01) |>
derive_seq.R:#' sdtm.oak:::is_seq_name("AESEQ")
derive_seq.R:#' sdtm.oak:::is_seq_name("AEseq")
derive_seq.R:#' sdtm.oak:::is_seq_name("AESEQUENCE")
dtc_create_iso8601.R:#' sdtm.oak:::iso8601_na(c("10", NA_character_))
dtc_create_iso8601.R:#' sdtm.oak:::zero_pad_whole_number(c(-1, 0, 1))
dtc_create_iso8601.R:#' sdtm.oak:::zero_pad_whole_number(c(-1, 0, 1, 10, 99, 100), n = 2)
dtc_create_iso8601.R:#' sdtm.oak:::zero_pad_whole_number(c(-1, 0, 1, 10, 99, 100), n = 3)
dtc_create_iso8601.R:#' sdtm.oak:::yy_to_yyyy(0:5)
dtc_create_iso8601.R:#' sdtm.oak:::yy_to_yyyy(2000:2005)
dtc_create_iso8601.R:#' sdtm.oak:::yy_to_yyyy(90:99)
dtc_create_iso8601.R:#' sdtm.oak:::yy_to_yyyy(1990:1999)
dtc_create_iso8601.R:#' sdtm.oak:::yy_to_yyyy(65:72)
dtc_create_iso8601.R:#' sdtm.oak:::yy_to_yyyy(1965:1972)
dtc_create_iso8601.R:#' sdtm.oak:::iso8601_two_digits(x)
dtc_create_iso8601.R:#' sdtm.oak:::iso8601_year(c("0", "1", "2", "50", "68", "69", "90", "99", "00"))
dtc_create_iso8601.R:#' sdtm.oak:::iso8601_year(c("67", "68", "69", "70"))
dtc_create_iso8601.R:#' sdtm.oak:::iso8601_year(c("1967", "1968", "1969", "1970"))
dtc_create_iso8601.R:#' sdtm.oak:::iso8601_year(as.character(0:50), cutoff_2000 = 25)
dtc_create_iso8601.R:#' sdtm.oak:::iso8601_year(as.character(1900:1950), cutoff_2000 = 25)
dtc_create_iso8601.R:#' sdtm.oak:::iso8601_mon(c(NA, "0", "1", "2", "10", "11", "12"))
dtc_create_iso8601.R:#' sdtm.oak:::iso8601_mon(c("13", "99", "100", "-1"))
dtc_create_iso8601.R:#' sdtm.oak:::iso8601_mon(mon)
dtc_create_iso8601.R:#' sdtm.oak:::iso8601_sec(c(NA, "0", "1", "10", "59", "99", "100"))
dtc_create_iso8601.R:#' sdtm.oak:::iso8601_truncate(x)
dtc_create_iso8601.R:#' sdtm.oak:::iso8601_truncate("-----T-:-:-", empty_as_na = TRUE)
dtc_create_iso8601.R:#' sdtm.oak:::iso8601_truncate("-----T-:-:-", empty_as_na = FALSE)
dtc_create_iso8601.R:#' sdtm.oak:::iso8601_truncate(
dtc_create_iso8601.R:#' sdtm.oak:::format_iso8601(m)
dtc_parse_dttm.R:#' sdtm.oak:::parse_dttm("2020", "y")
dtc_parse_dttm.R:#' sdtm.oak:::parse_dttm("2020-05", "y")
dtc_parse_dttm.R:#' sdtm.oak:::parse_dttm("2020-05", "y-m")
dtc_parse_dttm.R:#' sdtm.oak:::parse_dttm("2020-05-11", "y-m-d")
dtc_parse_dttm.R:#' sdtm.oak:::parse_dttm("2020 05 11", "y m d")
dtc_parse_dttm.R:#' sdtm.oak:::parse_dttm("2020  05  11", "y m d")
dtc_parse_dttm.R:#' sdtm.oak:::parse_dttm("2020  05  11", "y\\s+m\\s+d")
dtc_parse_dttm.R:#' sdtm.oak:::parse_dttm("2020      05     11", "y\\s+m\\s+d")
dtc_parse_dttm.R:#' sdtm.oak:::parse_dttm("2020-05-11 11:45", "y-m-d H:M")
dtc_parse_dttm.R:#' sdtm.oak:::parse_dttm("2020-05-11 11:45:15.6", "y-m-d H:M:S")
dtc_parse_dttm.R:#' sdtm.oak:::parse_dttm(c("2002-05-11 11:45", "-05-11 11:45"), "y-m-d H:M")
dtc_parse_dttm.R:#' sdtm.oak:::parse_dttm(c("2002-05-11 11:45", "-05-11 11:45"), "-m-d H:M")
dtc_parse_dttm.R:#' sdtm.oak:::parse_dttm(c("2002-05-11 11:45", "-05-11 11:45"), c("y-m-d H:M", "-m-d H:M"))
dtc_parse_dttm.R:#' sdtm.oak:::parse_dttm("05 feb 1985 12 55 02", "d m y H M S")
dtc_parse_dttm.R:#' sdtm.oak:::parse_dttm("12 55 02 05 feb 1985", "H M S d m y")
dtc_parse_dttm.R:#' sdtm.oak:::parse_dttm(c("2020-05-18", "2020-UN-18", "2020-UNK-UN"), "y-m-d")
dtc_parse_dttm.R:#' sdtm.oak:::parse_dttm(c("2020-05-18", "2020-UN-18", "2020-UNK-UN"), "y-m-d", na = "UN")
dtc_parse_dttm.R:#' sdtm.oak:::parse_dttm(c("2020-05-18", "2020-UN-18", "2020-UNK-UN"), "y-m-d", na = c("UN", "UNK"))
dtc_problems.R:#' dttm2 <- sdtm.oak:::add_problems(dttm, is_problem, dtc)
dtc_problems.R:#' sdtm.oak:::problems(dttm2)
dtc_problems.R:#' sdtm.oak:::any_problems(list(sdtm.oak:::parse_dttm("1980-06-18", "y-m-d")))
dtc_problems.R:#' sdtm.oak:::any_problems(list(sdtm.oak:::parse_dttm("1980-06-18", "ymd")))
dtc_problems.R:#' cap_matrix_date <- sdtm.oak:::parse_dttm(date, "y-m-d")
dtc_problems.R:#' cap_matrix_time <- sdtm.oak:::parse_dttm(time, "H:M:S")
dtc_problems.R:#' sdtm.oak:::any_problems(cap_matrices)
dtc_utils.R:#' sdtm.oak:::assert_dtc_fmt(c("ymd", "y m d", "dmy", "HM", "H:M:S", "y-m-d H:M:S"))
dtc_utils.R:#'   sdtm.oak:::assert_dtc_fmt("y years m months d days")
dtc_utils.R:#' sdtm.oak:::assert_dtc_format("ymd")
dtc_utils.R:#' sdtm.oak:::assert_dtc_format(c("ymd", "y-m-d"))
dtc_utils.R:#' sdtm.oak:::assert_dtc_format(list(c("ymd", "y-m-d"), "H:M:S"))
dtc_utils.R:#'   sdtm.oak:::assert_dtc_format("year, month, day")
dtc_utils.R:#' sdtm.oak:::assert_capture_matrix(m)
dtc_utils.R:#'   sdtm.oak:::assert_capture_matrix(character())
dtc_utils.R:#'   sdtm.oak:::assert_capture_matrix(matrix(data = NA_character_, nrow = 0, ncol = 0))
dtc_utils.R:#'   sdtm.oak:::assert_capture_matrix(matrix(data = NA_character_, nrow = 1))
dtc_utils.R:#' sdtm.oak:::complete_capture_matrix(matrix(data = NA_character_, nrow = 0, ncol = 0))
dtc_utils.R:#' sdtm.oak:::complete_capture_matrix(matrix(data = NA_character_, nrow = 1))
dtc_utils.R:#' # sdtm.oak:::complete_capture_matrix(m)
dtc_utils.R:#' # sdtm.oak:::complete_capture_matrix(m)
dtc_utils.R:#' # sdtm.oak:::complete_capture_matrix(m)
dtc_utils.R:#' sdtm.oak:::coalesce_capture_matrices(m_dates, m_times)
dtc_utils.R:#' sdtm.oak:::coalesce_capture_matrices(m_times, m_dates)
dtc_utils.R:#' sdtm.oak:::coalesce_capture_matrices(m_dates)
dtc_utils.R:#' sdtm.oak:::coalesce_capture_matrices(m_times)
oak_id_vars.R:#' sdtm.oak:::contains_oak_id_vars(oak_id_vars())
oak_id_vars.R:#' sdtm.oak:::contains_oak_id_vars(character())
oak_id_vars.R:#' sdtm.oak:::contains_oak_id_vars(c("oak_id", "raw_source"))
parse_dttm_fmt.R:#' sdtm.oak:::regex_or(c("jan", "feb"))
parse_dttm_fmt.R:#' paste0(sdtm.oak:::regex_or(c("jan", "feb"), .close = TRUE), r"{\d{2}}")
parse_dttm_fmt.R:#' sdtm.oak:::fmt_rg()
parse_dttm_fmt.R:#' sdtm.oak:::fmt_rg(mon = r"[\b\d|\d{2}]")
parse_dttm_fmt.R:#' sdtm.oak:::fmt_rg(na = "UNK")
parse_dttm_fmt.R:#' sdtm.oak:::fmt_rg(year_na = "UNK")
parse_dttm_fmt.R:#' sdtm.oak:::parse_dttm_fmt("ymd")
parse_dttm_fmt.R:#' sdtm.oak:::parse_dttm_fmt("H:M:S")
parse_dttm_fmt.R:#' sdtm.oak:::parse_dttm_fmt("ymd HMS")
parse_dttm_fmt.R:#' sdtm.oak:::parse_dttm_fmt("yymmdd HHMMSS")
parse_dttm_fmt.R:#' sdtm.oak:::parse_dttm_fmt("year y")
parse_dttm_fmt.R:#' sdtm.oak:::parse_dttm_fmt(
parse_dttm_fmt.R:#' sdtm.oak:::dttm_fmt_to_regex("y")
parse_dttm_fmt.R:#' sdtm.oak:::dttm_fmt_to_regex("y", anchored = FALSE)
parse_dttm_fmt.R:#' sdtm.oak:::dttm_fmt_to_regex("m")
parse_dttm_fmt.R:#' sdtm.oak:::dttm_fmt_to_regex("ymd")
parse_dttm_fmt.R:#' sdtm.oak:::dttm_fmt_to_regex("ymd HH:MM:SS")
recode.R:#' sdtm.oak:::index_for_recode(x = 1:5, from = c(2, 4))
recode.R:#' sdtm.oak:::recode(x,
recode.R:#' sdtm.oak:::recode(
recode.R:#' sdtm.oak:::recode(
```

### PR DESCRIPTION
**Pull request by ShiyuC**
This PR moves all the example sections for internal functions to unit tests.
Closes #63 

Please check off each taskbox as an acknowledgment that you completed the task
or check off that it is not relevant to your Pull Request. This checklist is
part of the Github Action workflows and the Pull Request will not be merged into
the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull z
Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the 
[tidyverse style guide](https://style.tidyverse.org/). Run 
`styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should
consider realistic data scenarios and edge cases, e.g. empty datasets, errors,
boundary cases etc. - See 
[Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you 
fully follow the 
[deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords
and families. Refer to the 
[categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the
`NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it
has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build oak site `pkgdown::build_site()` and check that all affected
examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by ShiyuC** [comment]
@ramiromagno @edgar-manukyan May I request your review of this MR? Thanks in advance

File: R/derive_blfl.R [hunk 1]
```diff
@@ -51,36 +38,9 @@ dtc_datepart <- function(dtc, partial_as_na = TRUE) {
 #'   seconds should be ignored (default is `TRUE`).
 #'
 #' @return Character vector containing ISO 8601 times.
-#' @export
```
Comment (edgar-manukyan) [inline R/derive_blfl.R:54]: Did we decide to not export this anymore?

**Review by edgar-manukyan — APPROVED** [review]
Thanks so much @ShiyuC for massive cleanup and awesome test 👍 I left one question for your kind consideration.

File: R/derive_blfl.R (hunk 1 above)
Comment (ShiyuC) [reply R/derive_blfl.R:54]: @edgar-manukyan thanks for checking! This function contains both "@keywords internal" and "@export". As I was trying to replace all the "sdtm:::" from the package, I included this one as well. Also, since `dtc_timepart()` is an internal function, I thought this might be the same. @ramiromagno could you confirm?

Comment (ramiromagno) [reply R/derive_blfl.R:54]: Hi @ShiyuC and @edgar-manukyan : these functions are part of a PR (see https://github.com/pharmaverse/sdtm.oak/commit/6abb5bac6a10fd6cfeaa936b7623d9feaa9aea49) by @kamilsi and co-authored by @rammprasad and @edgar-manukyan :) I'm not sure what the original intention was...

Comment (ramiromagno) [reply R/derive_blfl.R:54]: But I am guessing the idea was to have these functions as internal only.

**Review by ramiromagno — CHANGES_REQUESTED** [review]
@ShiyuC: great job! Thank you so much.

I think the only remaining bit is really to fix the test failures from the testing of `derive_blfl()`. But I am guessing Kamil or Ram are better equipped to give you feedback here as they worked on this function.

And also still on `derive_blfl()`, I would convert `expect_snapshot_error()` to `expect_snapshot(..., error = TRUE)` as `expect_snapshot_error()` is under "questioning" status (check its documentation).

## Test failures related to `derive_blfl()`

```R
==> Testing R file using 'testthat'

ℹ Loading sdtm.oak
[ FAIL 8 | WARN 0 | SKIP 0 | PASS 7 ]

── Failure (test-derive_blfl.R:47:3): derive_blfl sdmt_in validations work ─────
Snapshot of `derive_blfl(...)` has changed:
old vs new
- Required variable `DOMAIN` is missing
+ Required variable `DOMAIN` is missing in `sdtm_in`

* Run testthat::snapshot_accept('derive_blfl') to accept the change.
* Run testthat::snapshot_review('derive_blfl') to interactively review the change.

── Failure (test-derive_blfl.R:58:3): derive_blfl sdmt_in validations work ─────
Snapshot of `derive_blfl(...)` has changed:
old vs new
- Required variables `oak_id`, `raw_source` and `patient_number` are missing
+ Required variables `oak_id`, `raw_source`, and `patient_number` are missing in `sdtm_in`

* Run testthat::snapshot_accept('derive_blfl') to accept the change.
* Run testthat::snapshot_review('derive_blfl') to interactively review the change.

── Failure (test-derive_blfl.R:74:3): derive_blfl sdmt_in validations work ─────
Snapshot of `derive_blfl(...)` has changed:
old vs new
- Required variables `VSORRES`, `VSSTAT`, `VSTESTCD` and `VSDTC` are missing
+ Required variables `VSORRES`, `VSSTAT`, `VSTESTCD`, and `VSDTC` are missing in `sdtm_in`

* Run testthat::snapshot_accept('derive_blfl') to accept the change.
* Run testthat::snapshot_review('derive_blfl') to interactively review the change.
```

**Review by ramiromagno — APPROVED** [review]

File: R/derive_blfl.R (hunk 1 above)
Comment (ShiyuC) [reply R/derive_blfl.R:54]: Confirmed with Ram that this function should be internal

**Comment by ShiyuC** [comment]
> @ShiyuC: great job! Thank you so much.
> 
> I think the only remaining bit is really to fix the test failures from the testing of `derive_blfl()`. But I am guessing Kamil or Ram are better equipped to give you feedback here as they worked on this function.
> 
> And also still on `derive_blfl()`, I would convert `expect_snapshot_error()` to `expect_snapshot(..., error = TRUE)` as `expect_snapshot_error()` is under "questioning" status (check its documentation).
> 
> ## Test failures related to `derive_blfl()`
> ```r
> ==> Testing R file using 'testthat'
> 
> ℹ Loading sdtm.oak
> [ FAIL 8 | WARN 0 | SKIP 0 | PASS 7 ]
> 
> ── Failure (test-derive_blfl.R:47:3): derive_blfl sdmt_in validations work ─────
> Snapshot of `derive_blfl(...)` has changed:
> old vs new
> - Required variable `DOMAIN` is missing
> + Required variable `DOMAIN` is missing in `sdtm_in`
> 
> * Run testthat::snapshot_accept('derive_blfl') to accept the change.
> * Run testthat::snapshot_review('derive_blfl') to interactively review the change.
> 
> ── Failure (test-derive_blfl.R:58:3): derive_blfl sdmt_in validations work ─────
> Snapshot of `derive_blfl(...)` has changed:
> old vs new
> - Required variables `oak_id`, `raw_source` and `patient_number` are missing
> + Required variables `oak_id`, `raw_source`, and `patient_number` are missing in `sdtm_in`
> 
> * Run testthat::snapshot_accept('derive_blfl') to accept the change.
> * Run testthat::snapshot_review('derive_blfl') to interactively review the change.
> 
> ── Failure (test-derive_blfl.R:74:3): derive_blfl sdmt_in validations work ─────
> Snapshot of `derive_blfl(...)` has changed:
> old vs new
> - Required variables `VSORRES`, `VSSTAT`, `VSTESTCD` and `VSDTC` are missing
> + Required variables `VSORRES`, `VSSTAT`, `VSTESTCD`, and `VSDTC` are missing in `sdtm_in`
> 
> * Run testthat::snapshot_accept('derive_blfl') to accept the change.
> * Run testthat::snapshot_review('derive_blfl') to interactively review the change.
> ```

Confirmed that the failed test will be addressed in a separate PR.

**Comment by ShiyuC** [comment]
@rammprasad I'm still not authorized to merge this PR after all conversations and conflicts are resolved.

**Comment by galachad** [comment]
> @rammprasad I'm still not authorized to merge this PR after all conversations and conflicts are resolved.

I will merge this PR, we can check in another one why.

**Comment by kamilsi** [comment]
> I think the only remaining bit is really to fix the test failures from the testing of `derive_blfl()`. But I am guessing Kamil or Ram are better equipped to give you feedback here as they worked on this function.

Hi @ramiromagno, the changes in test behavior are not justified by any codebase changes in this PR. Perhaps admiraldev was updated on the machine where the tests were run, allowing it to identify the object where the problem is detected. Regardless, these are good changes and should be accepted using testthat::snapshot_accept('derive_blfl').

> Confirmed that the failed test will be addressed in a separate PR.

@ShiyuC which issue is that?
